### ISSUE TITLE
Remove/replace tests against graphhopper.com

### ISSUE THREAD

**Issue by easbar** [graphhopper/graphhopper#1986]
`GraphHopperMatrixIT` and `GraphHopperWebIT` are running against the public GH service, but I think this not ideal. If they fail its often because of some data changes and nothing can be done here in the open source project except fixing the expected values. 

What should be tested (instead):

* test that `GraphHopper(Matrix)Web` is **sending** the right requests (no need to actually send the requests to graphhopper.com for this). Either just unit test the methods that create the request from `GH(Matrix)Request` or do as its done right now but send the requests to some test server on localhost.

* test the way the (hosted) GH server answers certain requests (but that should not happen here in the open source project).

**Comment by easbar** [comment]
I just fixed the expected distance here again:

And it seems odd that only the distance (but not the weight) changed. Here is the route: 

https://graphhopper.com/maps/?point=51.467697%2C-0.090637&point=51.521241%2C-0.171833&locale=en-us&vehicle=car&weighting=fastest&turn_costs=true&use_miles=false&layer=Omniscale

![image](https://user-images.githubusercontent.com/17603532/79213889-3e87fa80-7e4a-11ea-8b50-c631e21390e5.png)

Note that the route looks a bit weird in a few places, for example just before the end:

![image](https://user-images.githubusercontent.com/17603532/79214065-78f19780-7e4a-11ea-8c86-6af00d2155fa.png)

**Comment by easbar** [comment]
Seems to be a mapping error, take a look at the different layers:

![image](https://user-images.githubusercontent.com/17603532/79214199-b1917100-7e4a-11ea-93c1-02f94503165e.png)

![image](https://user-images.githubusercontent.com/17603532/79214226-beae6000-7e4a-11ea-924b-7c55e595c5aa.png)

![image](https://user-images.githubusercontent.com/17603532/79214266-d4238a00-7e4a-11ea-9dad-085eea38073c.png)

**Comment by karussell** [comment]
I removed the matrix tests that goes to graphhopper.com: https://github.com/graphhopper/graphhopper/commit/a5ca5ff4f43012f19db32285b733d2f02838c37d

The other tests I would not want to remove but to replace them with local queries so that we know that the client is "really working".

**Comment by easbar** [comment]
> but to replace them with local queries 

Isn't this the same as we currently do with `RouteResourceTest` etc. but instead of using `DropwizardRule#app.client()` we use `GraphHopperWeb` to send the request?

**Comment by karussell** [comment]
> Isn't this the same as we currently do with RouteResourceTest etc. but instead of using DropwizardRule#app.client() we use GraphHopperWeb to send the request?

Yes, probably we should use our client to do these requests :)

**Comment by easbar** [comment]
> Yes, probably we should use our client to do these requests :)

Hm, not sure because we want to test the server functionality independent from the client (because the client-hc is not the only way to query the web api). But what I wanted to say is that having a test that does use client-hc against localhost should work just like we did the RouteResource tests so far?

**Comment by karussell** [comment]
Ok

> But what I wanted to say here is that having a test that does use client-hc against localhost should work just like we did the RouteResource tests so far?

Yes

**Comment by easbar** [comment]
Converting `GraphHopperWebIT` is not 100% straight forward as it runs queries in very different locations and we would need multiple test maps to cover them all. Then again `GraphHopperWebIT` should not even test anything too specific and I think the tests can either be removed or be converted so they are all running on one of the test maps we already use in the repo.

**Comment by karussell** [comment]
Yeah, moving everything into one region is fine. I'll take over here and see if I can avoid removing too much coverage.
